File: python/1-chat-with-copilot/after/StarWarsCopilot/Program.py
from __future__ import annotations

import logging

from openai import AsyncAzureOpenAI

from LLMOptions import LLMOptions

logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
logger = logging.getLogger("starwars-copilot")

async def main() -> None:
    client = AsyncAzureOpenAI(
        api_key=LLMOptions.api_key,
        azure_endpoint=LLMOptions.endpoint,
        api_version="2024-10-21",
    )

    while True:
        user_input = input("User > ").strip()
        if not user_input:
            break

        completion = await client.chat.completions.create(
            model=LLMOptions.model,
            messages=[{"role": "user", "content": user_input}],
        )

        text = completion.choices[0].message.content or ""
        logger.info("response id=%s", completion.id)
        print(f"Assistant > {text}")


if __name__ == "__main__":
    import asyncio

    asyncio.run(main())

File: csharp/1-chat-with-copilot/after/StarWarsCopilot/Program.cs
using System.ClientModel;
using Azure.AI.OpenAI;

using Microsoft.Extensions.AI;
using Microsoft.Extensions.Logging;

using StarWarsCopilot;


// Create a logger factory
var factory = LoggerFactory.Create(builder => builder.AddConsole()
                                                     .SetMinimumLevel(LogLevel.Trace));


var client = new AzureOpenAIClient(new Uri(LLMOptions.Endpoint),
                                   new ApiKeyCredential(LLMOptions.ApiKey));

var innerClient = client.GetChatClient(LLMOptions.Model).AsIChatClient();

var chatClient = new ChatClientBuilder(innerClient)
                        .UseLogging(factory)
                        .Build();

// Initiate a back-and-forth chat
while (true)
{
    // Collect user input
    Console.Write("User > ");
    var userInput = Console.ReadLine();

    // End processing if user input is null or empty
    if (string.IsNullOrWhiteSpace(userInput))
        break;

    // Get the response from the AI
    var result = await chatClient.GetResponseAsync(userInput);

    // Print the results
    Console.WriteLine("Assistant > " + result.Messages.Last()?.Text);
}